aspnetcore/blazor/common/samples/3.x/BlazorSample_Server/Pages/forms-and-validation/FormExample4.razor

```diff
@@ -38,32 +28,46 @@
 </EditForm>
 
 @code {
-    private CustomValidation customValidation;
-    private Starship starship = new Starship() { ProductionDate = DateTime.UtcNow };
+    private EditContext editContext;
+    private Holodeck holodeck = new Holodeck();
+    private ValidationMessageStore validationMessageStore;
```

Comment:
This is like a super weird nit, but I wonder if it's better to make these `readonly`. Particularly the store, because you really don't want to new it up at random points of time and have no way to clear messages from the old instance. 


For instance, you could do this, and only use the OnInitialized to configure the validation requested

```suggestion
    private readonly Holodeck holodeck = new();
    private readonly EditContext editContext = new(holodeck);
    private readonly ValidationMessageStore validationMessageStore = new(editContext);
```